client/platform/desktop/backend/native/common.ts

```diff
@@ -163,24 +164,28 @@ async function _findCSVTrackFiles(searchPath: string) {
 /**
  * locate json track file in a directory
  * @param path path to a directory
- * @returns absolute path to json file or empty string
+ * @returns object containing trackAbsPath and metaPath if it exists
  */
-async function _findJsonTrackFile(basePath: string): Promise<string> {
+async function _findJsonAndMetaTrackFile(basePath: string): Promise<
+  {trackFileAbsPath: string; metaFileAbsPath?: string}> {
   const contents = await fs.readdir(basePath);
   const jsonFileCandidates: string[] = [];
+  let metaFileAbsPath;
   await Promise.all(contents.map(async (name) => {
     const fullPath = npath.join(basePath, name);
     if (JsonTrackFileName.test(name)) {
       const statResult = await fs.stat(fullPath);
       if (statResult.isFile()) {
         jsonFileCandidates.push(fullPath);
       }
+    } else if (name === JsonMetaFileName) {
+      metaFileAbsPath = fullPath;
```

Comment:
I don't think this works for most use cases.

Config downloaded from web is named `"dataset_name.config.json"`, so desktop will not attempt to parse it?